File: BackupUtility/Classes/ConfirmationChecker.cs
namespace ZCopy
{
    public class ConfirmationChecker : IConfirmationChecker
    {
        public delegate bool ConfirmationRequest(string theInfo);

        private readonly ConfirmationRequest _ConfirmationRequest;
        private readonly bool _requestConfirm;

        public ConfirmationChecker(bool requestConfirm)
        {
            _requestConfirm = requestConfirm;
        }

        public bool GetConfirmation(string aTarget)
        {
            if (_requestConfirm && _ConfirmationRequest != null)
                return _ConfirmationRequest(aTarget);
            return true;
        }
    }
}

File: BackupUtility/Classes/ConfirmationChecker.vb
Public Class ConfirmationChecker
    Implements IConfirmationChecker

    Public Delegate Function ConfirmationRequest(ByVal theInfo As String) As Boolean

    Private ReadOnly _ConfirmationRequest As ConfirmationRequest
    Private ReadOnly _requestConfirm As Boolean

    Public Sub New(requestConfirm As Boolean)
        _requestConfirm = requestConfirm
    End Sub

    Public Function GetConfirmation(aTarget As String) As Boolean Implements IConfirmationChecker.GetConfirmation
        If _requestConfirm AndAlso _ConfirmationRequest IsNot Nothing Then
            Return _ConfirmationRequest(aTarget)
        End If
        Return True
    End Function
End Class
